### ISSUE TITLE
Update from Torchscript to `torch.export`

### ISSUE THREAD

**Issue by SkafteNicki** [Lightning-AI/pytorch-lightning#21293]
### Outline & Motivation

Torchscript is deprecated in the most recent releases of Pytorch
https://docs.pytorch.org/docs/stable/jit.html
Instead `torch.export` is the recommended way to go:
https://docs.pytorch.org/docs/stable/export.html

The most obvious change that need to be made is to `LightningModule.to_torchscript` which either:
* keep function but update internals to use the new `torch.export` functionality
* deprecate/remove and introduce a new function called `export` or similar that use new functionality
* deprecate/remove and let users manually export

In addition to this we need to update documentation etc. since there are still multiple references to `torch.jit`. See this hotfix for when torch v2.9 was released: https://github.com/Lightning-AI/pytorch-lightning/pull/21292

cc @lantiga @justusschock

**Comment by GdoongMathew** [comment]
I'm not so sure about the 1st option - since `torch.export` doesn't produce the same `ScriptModule` object as `to_torchscript`, simply replacing the implementation with `torch.export` might cause confusion.

I'd prefer the second approach, maybe introducing a method like `to_exportedprogram`, so we could take advantage of the existing exporting mechanisms, (like `example_input_array` and others).

**Comment by SkafteNicki** [comment]
Agree that option 1 is not optimal but just wanted to list all options for sake of completeness. Also in favor of option 2.
